src/clr-angular/forms/styles/_checkbox.clarity.scss

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+// This software is released under MIT license.
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@include exports('forms.checkbox') {
+    $mark-half-size: $clr-forms-checkbox-mark-size / 2;
+    .clr-checkbox-wrapper {
+        @include form-flatten-validate-text();
+
+        //Hide the default checkbox behind the pseudo elements
+        //Opacity is set to 0 so that the input remains accessible
+        input[type="checkbox"] {
+            @include checkbox-radio-input-styles($clr-forms-checkbox-size);
+        }
+
+        label {
+            @include checkbox-radio-label-styles($clr-forms-checkbox-label-height, $clr-forms-checkbox-padding-left, $clr-forms-checkbox-label-color);
+        }
+
+        //Checkbox base
+        input[type="checkbox"] + label::before {
+            @include checkbox-radio-shared-inactive(
+                $clr-forms-checkbox-height, 
+                $clr-forms-checkbox-top, 
+                0, 
+                $clr-forms-border-color);
+            border-radius: $clr-global-borderradius;
+        }
+
+        input[type="checkbox"]:focus + label::before {
+            @include include-outline-style-form-fields();
+        }
+
+        //Checkmark
+        input[type="checkbox"] + label::after {
+            position: absolute;
+            content: '';
+            display: none;
+            height: $mark-half-size + $clr-rem-1px;
+            width: $clr-forms-checkbox-mark-size;
+            border-left: $clr-forms-checkbox-border-style;
+            border-bottom: $clr-forms-checkbox-border-style;
+            top: ($clr-forms-baseline * 4 - $clr-checkbox-radio-height) / 2;
+            left: $mark-half-size;
+            transform: translate(0, $mark-half-size) rotate(-45deg);
+            border-color: $clr-forms-checkbox-mark-color;
+        }
+
+        // @TODO How to get checked state without a parent class?
+        input[type="checkbox"]:checked + label::before {
+            background: $clr-forms-checkbox-background-color;
+            border: none;
+        }
+
+        input[type="checkbox"]:checked + label::after {
+            display: inline-block;
+        }
+
+        input[type="checkbox"].clr-indeterminate + label::before {
+            border: $clr-default-borderwidth solid $clr-forms-checkbox-indeterminate-border-color;
+        }
+        input[type="checkbox"].clr-indeterminate + label::after {
+            border-left: none;
+            border-bottom-color: $clr-forms-checkbox-indeterminate-border-color;
+            display: inline-block;
+            transform: translate(0, $mark-half-size);
+        }
+
+        label > input[type="checkbox"]::after {
```

Comment:
Something doesnt look right here. I dont think the `input` element has an `::after` pseudo element, no?

Is this `input[type="checkbox"] > label::after` instead of what you have here currently?

Reply:
Wait the styles look repeated on line 35. Line 35 has the correct selector. This one looks incorrect.

Reply:
https://github.com/vmware/clarity/pull/2068/files#diff-c6fa1884c13e1bcc0567ed5b96b0c75cR35

Reply:
This is leftover from attempting to support inputs inside of labels, and I'll remove it for now.